src/models/user.model.ts

```diff
@@ -0,0 +1,147 @@
+import { DynamoDbSchema, DynamoDbTable, QueryOptions, ScanOptions } from '@aws/dynamodb-data-mapper'
+import mapper from '../db/dynamoDbClient'
+import { v4 as uuid } from 'uuid'
+import { Number } from 'aws-sdk/clients/iot'
+import MapBoxService from '../services/mapbox'
+
+const USER_NOT_FOUND = 'Not able to find user: '
+const USER_NOT_SAVED = 'Not able to save user: '
+const USER_NOT_DELETED = 'Not able to delete user: '
+
+class User {
+  id: String | undefined
+  name: String | undefined
+  dob: Date | undefined
+  address: String | undefined
+  description: String | undefined
+  createdAt: Date | undefined
+  updateAt: Date | undefined
+  imageUrl: String | undefined
+
+  constructor(name?: String,
+    dob?: Date,
+    address?: String,
+    description?: String,
+    imageUrl?: String) {
+    const currentDate: Date = new Date()
+    this.id = uuid()
+    this.name = name
+    this.dob = dob
+    this.address = address
+    this.description = description
+    this.createdAt = currentDate
+    this.updateAt = currentDate
+    this.imageUrl = imageUrl
+  }
+  save(): Promise<this> {
+    return mapper.put(this).catch((err) => Promise.reject(USER_NOT_SAVED + err))
```

Comment:
How would you prevent internal details on errors from the database from leaking the domain of the GraphQL API?

Reply:
Catching the internal errors and converted to predefined errors message which would be the one displayed to the user or upper layers. GraphQL errors should only be a concern of the monitoring process, so those detailed error messages would be logged to any logging and monitoring strategy chosen (cloudwatch, graylogm splunk, etc)